lib/rules/jsx-filename-extension.js

```diff
@@ -32,9 +32,12 @@ module.exports = function(context) {
   return {
 
     JSXElement: function(node) {
-      var allowedExtensions = getExtensionsConfig();
       var filename = context.getFilename();
+      if (filename === '<text>') {
```

Comment:
Sadly the only place it exists is https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L220